### DISCUSSION TITLE
output.shoutcast Broken pipe in write()

### DISCUSSION THREAD

**Discussion by SebaLedesma** [savonet/liquidsoap#2304] [User Support]
I'm using liquidsoap version 1.4.1-1ubuntu0.20.04.1.
I cant make output.shoutcast work. I've tried with two different shoutcast servers. I've also checked the the first server with mpd and it works fine.
In the other hand output.icecast works fine.

**To Reproduce**
Create a simple script like this:
output.shoutcast(%mp3(bitrate=320), host = "62.210.129.76", port = 1234, password = "***", input.alsa())
or execute
$ liquidsoap 'output.shoutcast(%mp3(bitrate=320), host = "62.210.129.76", port = 1234, password = "***", input.alsa())'

Also tested (and failed) blank 
$ liquidsoap 'output.shoutcast(%mp3(bitrate=320), host = "62.210.129.76", port = 1234, password = "***", blank())'
and a single file:
$ liquidsoap 'source = single("test.wav"); output.shoutcast(%mp3, host = "62.210.129.76", port = 1234, password = "***", blank())'

**Expected behavior**
Connect and stream.
Insted it connects and fails, reconnect after 3 seconds and fails.
Log:
2022/03/18 20:36:32 [gstreamer.loader:3] Loaded GStreamer 1.16.2 0
2022/03/18 20:36:32 [dynamic.loader:3] Could not find dynamic module for fdkaac encoder.
2022/03/18 20:36:32 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz master.
2022/03/18 20:36:32 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2022/03/18 20:36:32 [frame:3] Targetting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2022/03/18 20:36:32 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2022/03/18 20:36:32 [sandbox:3] Sandboxing disabled
2022/03/18 20:36:32 [video.converter:3] Using preferred video converter: gavl.
2022/03/18 20:36:32 [audio.converter:3] Using samplerate converter: ffmpeg.
2022/03/18 20:36:32 [output(dot)shoutcast:3] Connecting mount sid#1 for @62.210.129.76...
2022/03/18 20:36:32 [input.alsa_9125:2] Falling back on interleaved S16LE
2022/03/18 20:36:33 [output(dot)shoutcast:3] Connection setup was successful.
2022/03/18 20:36:33 [clock.wallclock_alsa:3] Streaming loop starts, synchronized by active sources.
2022/03/18 20:36:33 [map_metadata_9130:3] Inserting missing metadata.
2022/03/18 20:36:34 [output(dot)shoutcast:2] Error while sending data: could not write data to host: Broken pipe in write()!
2022/03/18 20:36:34 [output(dot)shoutcast:3] Closing connection...
2022/03/18 20:36:34 [output(dot)shoutcast:3] Will try to reconnect in 3.00 seconds.
2022/03/18 20:36:38 [output(dot)shoutcast:3] Connecting mount sid#1 for @62.210.129.76...
2022/03/18 20:36:44 [output(dot)shoutcast:3] Connection setup was successful.
2022/03/18 20:36:45 [output(dot)shoutcast:2] Error while sending data: could not write data to host: Broken pipe in write()!
2022/03/18 20:36:45 [output(dot)shoutcast:3] Closing connection...
2022/03/18 20:36:45 [output(dot)shoutcast:3] Will try to reconnect in 3.00 seconds.
2022/03/18 20:36:49 [output(dot)shoutcast:3] Connecting mount sid#1 for @62.210.129.76...

**Version details**
 - Ubuntu 20.04, kernel 4.9.277-83 #1 SMP PREEMPT Mon Feb 28 15:16:26 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
 - liquidsoap version 1.4.1-1ubuntu0.20.04.1

**Install method**
sudo apt install liquidsoap

**Comment by toots** [comment]
Have you checked the ports open on the shoutcast server? Shoutcast uses the configured `port` as well as `port+1` for source connections so, in this script, `1234` and `1235`.

**Reply by SebaLedesma** [reply]
Server seems fine as I used another tool to test it: Music Player Daemon.
Also I've tested with two different servers, in both cases MPD can stream, Liquidsoap presents the 'broken pipe' error.

As alternative I've tried output.icecast in shoutcast mode, but I was unable to make it work (I didnt found any example).

**Reply by SebaLedesma** [reply]
I've managed to make work 'output.icecast' with shoutcast server, just add the parameter protocol='icy'
In the previous example:
liquidsoap 'output.icecast(%mp3, host = "62.210.129.76", port = 1234, password = "hackme!", protocol="icy", input.alsa())'

**Reply by toots** [reply]
Did it start working?

**Comment by SebaLedesma** [comment]
output.shoutcast still has the problem, but output.icecast configuired with protocol="icy" works.

**Reply by toots** [reply]
Yeah, that's how `output.shoutcast` is written, too, using `output.icecast`

We've just had another report on slack about a similar error. I'll have a look at this when I get a moment.
